homeworks/constantin/1-halfedge/README.md

```diff
@@ -0,0 +1,56 @@
+# Homework 1: Implement the Halfedge Structure
+
+**Task 1. Build Halfedge-based mesh structure in the [`buildMesh`](./src/geometry/halfedge.ts) method, then render the mesh using the provided [software rasterizer](./src/).**
+
+![](./assets/expect.png)
+
+- _Hint: Don't work directly on a complex mesh, but try to start with a single triangle._
+- Think about how to turn the learned data structure to code.
+- Read blender's [developer documentation](https://wiki.blender.org/wiki/Source/Modeling/BMesh/Design), try to answer these questions:
+- What is the blender's internal mesh data structure?
+- How it works and what are the differences compare to Halfedge-based mesh strcture?
+
+Write the answers here:
+
+```
+Blenders internal data structure is called BMesh. A BMesh stores Faces,Loops,Edges and Vertices for each Mesh (Main Element structures). Each Loop corresponds to an edge, but there can be more than one loop per edge. Each Loop stores how these Main element structures are connected, and therefore a BMesh can represent also non-manifolds. A halfedge however can only represent manifold meshes.
```

Comment:
Missing the most important part of how does Bmesh addresses non-manifold meshes.